src/data/checkfiles/default_checkfiles/unused-metadata.js

```diff
@@ -0,0 +1,148 @@
+    } else {
+      let totalMetaQuery = "SELECT META_DATA_ATTR_ID";
+
+      const totalMetaResp = await axios({
+        url: `${this.restApiLocation}/query`,
+        method: "GET",
+        headers: {
+          Authorization: localStorage.getItem("zmt-token"),
+        },
+        params: {
+          query: totalMetaQuery,
+          type: "general",
+          limit: 999999999,
+          offset: 0,
+          "case-sensitive": 0,
+        },
+      }).catch((e) => {});
+
+      if (totalMetaResp && totalMetaResp.data) {
+        totalMetaResp.data._embedded.forEach((row) => {
+          totalMeta.add(row[0]);
+        });
+      }
+
+      console.log("totalMeta:", totalMeta);
+
+      const getMeta = async (query) => {
+        const resp = await axios({
+          url: `${this.restApiLocation}/query`,
+          method: "GET",
+          headers: {
+            Authorization: localStorage.getItem("zmt-token"),
+          },
+          params: {
+            query: query,
+            type: "general",
+            limit: 999999999,
+            offset: 0,
+            "case-sensitive": 0,
+          },
+        }).catch((e) => {
+          console.log(e);
+        });
+
+        let useSet = new Set();
+        if (resp && resp.data) {
+          console.log("resp.data", resp.data);
+
+          // turn the first item in each row into a set
+          resp.data._embedded.forEach((row) => {
+            useSet.add(row[0]);
+          });
+        }
+
+        return useSet;
+      };
+
+      // find all metadata associated with users
+      let userMetaQuery = "SELECT META_USER_ATTR_ID, USER_ID";
+      let userMetaSet = await getMeta(userMetaQuery);
+
+      let dataObjMetaQuery = "SELECT META_DATA_ATTR_ID, DATA_ID";
+      let dataObjMetaSet = await getMeta(dataObjMetaQuery);
+
+        let rescMetaQuery = "SELECT META_RESC_ATTR_ID, RESC_ID";
+        let rescMetaSet = await getMeta(rescMetaQuery);
+
+        let collMetaQuery = "SELECT META_COLL_ATTR_ID, COLL_ID";
+        let collMetaSet = await getMeta(collMetaQuery);
+
+        numUnusedMeta = totalMeta.size - userMetaSet.size - dataObjMetaSet.size - rescMetaSet.size - collMetaSet.size;
+        console.log("numUnusedMeta", numUnusedMeta);
+    }
+
+    result.message = []
+
+    if (numUnusedMeta > 0) {
+
+        result.message.push(<span key="unusedMeta1">There are {numUnusedMeta} unused metadata.</span>);
+    } else {
+        result.message.push(<span key="unusedMeta2">All metadata is used.</span>);
+    }
+
+    result.status = (numUnusedMeta > 0 || specificWarning) ? "warning" : "healthy";
+
+    if (specificWarning) {
+
+        // " Additionally, to make this check faster, please add this specific query: " + <Chip label={specificQuery} />;
+
+        result.message.push(<span key="unusedMeta3"> Additionally, to make this check faster, please add this specific query: <Chip label={specificQuery} /></span>);
+
+
+    } else {
+
```

Comment:
what happens here?

Reply:
I was going to do the logic differently at first, but ended up changing it. Haven't gotten around to code cleanup yet at the time of review haha